packages/enzyme/src/ShallowWrapper.js

```diff
@@ -310,78 +311,60 @@ class ShallowWrapper {
         const { state } = instance;
         const prevProps = instance.props || this[UNRENDERED].props;
         const prevContext = instance.context || this[OPTIONS].context;
-        const nextProps = { ...prevProps, ...props };
         const nextContext = context || prevContext;
         if (context) {
           this[OPTIONS] = { ...this[OPTIONS], context: nextContext };
         }
         this[RENDERER].batchedUpdates(() => {
+          // When shouldComponentUpdate returns false we shouldn't call componentDidUpdate.
+          // so we spy shouldComponentUpdate to get the result.
           let shouldRender = true;
-          // dirty hack:
-          // make sure that componentWillReceiveProps is called before shouldComponentUpdate
-          let originalComponentWillReceiveProps;
+          let spy;
           if (
             !this[OPTIONS].disableLifecycleMethods &&
             instance &&
-            typeof instance.componentWillReceiveProps === 'function'
+            typeof instance.shouldComponentUpdate === 'function'
           ) {
-            instance.componentWillReceiveProps(nextProps, nextContext);
-            originalComponentWillReceiveProps = instance.componentWillReceiveProps;
-            instance.componentWillReceiveProps = () => {};
+            spy = spyMethod(instance, 'shouldComponentUpdate');
+          }
+          if (props) this[UNRENDERED] = cloneElement(adapter, this[UNRENDERED], props);
+          this[RENDERER].render(this[UNRENDERED], nextContext);
+          if (spy) {
+            shouldRender = spy.getLastReturnValue();
+            spy.restore();
           }
-          // dirty hack: avoid calling shouldComponentUpdate twice
-          let originalShouldComponentUpdate;
           if (
+            shouldRender &&
             !this[OPTIONS].disableLifecycleMethods &&
-            instance &&
-            typeof instance.shouldComponentUpdate === 'function'
+            instance
           ) {
-            shouldRender = instance.shouldComponentUpdate(nextProps, state, nextContext);
-            originalShouldComponentUpdate = instance.shouldComponentUpdate;
-          }
-          if (shouldRender) {
-            if (props) this[UNRENDERED] = cloneElement(adapter, this[UNRENDERED], props);
-            if (originalShouldComponentUpdate) {
-              instance.shouldComponentUpdate = () => true;
-            }
-
-            this[RENDERER].render(this[UNRENDERED], nextContext);
-
-            if (originalShouldComponentUpdate) {
-              instance.shouldComponentUpdate = originalShouldComponentUpdate;
-            }
             const lifecycles = getAdapterLifecycles(adapter);
+            // getSnapshotBeforeUpdate
+            let snapshot;
             if (
-              !this[OPTIONS].disableLifecycleMethods &&
-              instance
+              lifecycles.getSnapshotBeforeUpdate
```

Comment:
this no longer seems to be checking `disableLifecycleMethods`?

Reply:
Checking `disableLifecycleMethods` is here.

https://github.com/airbnb/enzyme/pull/1192/files#diff-fa27c82ee4fde075bae6173848e17c82R338